debezium-core/src/main/java/io/debezium/config/Field.java

```diff
@@ -393,14 +365,14 @@ public static ConfigDef group(ConfigDef configDef, String groupName, Field... fi
                 for (int i = 0; i != fields.length; ++i) {
                     Field f = fields[i];
                     configDef.define(f.name(), f.type(), f.defaultValue(), null, f.importance(), f.description(),
-                            groupName, i + 1, f.width(), f.displayName(), f.dependents(), null);
+                            groupName, i + 1, f.width(), f.displayName(), null, null);
```

Comment:
The build errors are related to those changes.
You need to remove that field from the method call completely! 
Then you will use the method that has Recommender but no Dependents config (see org.apache.kafka.common.config.ConfigDef line 190) :
```
    public ConfigDef define(String name, Type type, Object defaultValue, Validator validator, Importance importance, String documentation,
                            String group, int orderInGroup, Width width, String displayName, Recommender recommender)
```

Reply:
```suggestion
                            groupName, i + 1, f.width(), f.displayName(), null);
```

Reply:
This needs to be done on some other lines as well.

Reply:
If we do that 
`groupName, i + 1, f.width(), f.displayName(), null);`
we run into ambiguous method call. One method takes List<String> as last argument and the other takes Recommender. 
Any idea?

Reply:
I'm not sure I understood your suggested change correctly, but if you meant that we can just pass null as the recommender value, we can always do something a bit nasty:
`Recommender recommender = null;
                configDef.define(f.name(), f.type(), f.defaultValue(), null, f.importance(),
                    f.description(), groupName, orderInGroup, f.width(), f.displayName(), (ConfigDef.Recommender) recommender);` 

Or maybe I didn't get it.

Reply:
There are more than 2. The old implementation uses the method with (..., dependents, recommender), you want to use (..., recommender) now.

Reply:
The result should look similar to:

`configDef.define(f.name(), f.type(), f.defaultValue(), f.importance(), f.description(), 
    groupName, orderInGroup, f.width(), f.displayName());`

Reply:
One question. 
I removed these methods cause they had the dependents:
```java
public void forEachMissingDependent(Consumer<String> consumer) {
            fieldsByName.values().stream()
                    .map(Field::dependents)
                    .flatMap(Collection::stream)
                    .filter(Predicates.not(fieldsByName::containsKey))
                    .forEach(consumer);
}
public void forEachTopLevelField(Consumer<Field> consumer) {
            Collection<String> namesOfDependents = fieldsByName.values().stream()
                    .map(Field::dependents)
                    .flatMap(Collection::stream)
                    .collect(Collectors.toSet());
            fieldsByName.values().stream().filter(f -> !namesOfDependents.contains(f.name())).forEach(consumer);
}
```
Now I'm not sure if that was correct because some tests fail. Should I somehow pass the recommender instead of dependents?

Reply:
I think removing is okay, imo. I think you have to fix the calling code. But I have to review the callers myself again.